### ISSUE TITLE
Add phasing capability

### ISSUE THREAD

**Issue by apriha** [apriha/lineage#2]
Add the capability to phase the chromosomes of a family group. This feature should identify the alleles that each child inherited from each parent, which in turn identifies the chromosomes of each parent (i.e., the grandparental source of the DNA).

Principally, this feature should use the results of [`find_shared_dna`](https://apriha.github.io/lineage/lineage.html#lineage.Lineage.find_shared_dna) so that the phasing algorithm knows the DNA that is identical by descent vs. identical by state - both pieces of information would be very helpful in performing the phasing, especially when data from additional family members are available so that more comparisons can be performed.

If only one parent's data is available, Whit Athey developed an algorithm to perform the phasing with data from the parent and multiple siblings: http://www.jogg.info/pages/62/files/Athey.pdf . In addition to helping with phasing when there are ambiguities, this technique can also be used to reverse engineering the DNA of a missing parent.

**Comment by apriha** [comment]
Consider integrating https://github.com/poruloh/Eagle

**Comment by ebacherdom** [comment]
> Consider integrating https://github.com/poruloh/Eagle

This only seems useful if no familial DNA is available - IBD gives a much more conclusive result for phasing than statistical methods.

**Comment by apriha** [comment]
@ebacherdom, I agree. Like discussed above, I think using the results of `find_shared_dna` would help with this, especially when more comparisons of individuals in a family group are available. Formally, I think this is a [constraint satisfaction](https://en.wikipedia.org/wiki/Constraint_satisfaction) problem.

**Comment by rjsdotorg** [comment]
For those who have done visual phasing on grandparents, the manual (no one has been able to automate it yet) assignments gleaned from associations with other DNA test company users is also a significant source of information for the problem.

**Comment by rjsdotorg** [comment]
The Lazarus tool at Gedmatch.com uses 3 groups for direct descent, exclusion (spouse's family), and inclusion (target's relatives).
Their algorithm is not published however.

> 
> **Group Zero** is optional - It should preferably contain only the spouse (other parent of group 1) of the Lazarus target. If the spouse's DNA is not available, then a list of kit numbers of people related to ancestors of the spouse may be used here. If the children (in group one) of the Lazarus target are half-siblings (one parent is different), then all of this group zero section should be left empty. The purpose of this section is to remove any DNA segment from the Lazarus target if it is found to match between group zero and group 2, since that segment will be considered to be endogamous.
> **Group One** - at least one entry is required. This is a list of kit numbers for available descendants of the target Lazarus person we wish to deduce the atDNA. That will be children and/or grandchildren. It is not necessary to include grandchildren if their parent (the child of the Lazarus target) is included.
> **Group Two** - at least one entry is required. This is a list of kit numbers for available remaining relatives of the target Lazarus person we wish to deduce the at DNA. This will be brother/sisters, aunts/uncles, parents, cousins of the Lazarus target.
